helpers/analysis.js

```diff
@@ -0,0 +1,159 @@
+// analyze distance between two locations
+var analyzeDistance = function(i, j) {
+  if ((Math.abs(i.latitude - j.latitude)) < 0.005 && (Math.abs(i.longitude - j.longitude)) < 0.005) {
+    return 0;
+  } else if ((Math.abs(i.latitude - j.latitude)) < 0.01 && (Math.abs(i.longitude - j.longitude)) < 0.01) {
+    return 1;
+  } else if ((Math.abs(i.latitude - j.latitude)) < 0.02 && (Math.abs(i.longitude - j.longitude)) < 0.02) {
+    return 2;
+  } else {
+    return 3;
+  }
+};
+
+
+// join two analyzed locations together
+var joinLocation = function(array, i, j) {
+  // check if the element is already included
+  array.forEach(function(element) {
+    var res = element.result.name.split("=");
+    if (i.name === res[res.length - 1]) {
+      // console.log(i.name);
+      array.push({
+        result: {
+          name: element.result.name + "=" + j.name,
+          latitude: element.result.latitude + "=" + j.latitude,
+          longitude: element.result.longitude + "=" + j.longitude,
+          address1: element.result.address1 + "=" + j.address1,
+          address2: element.result.address2 + "=" + j.address2
+        }
+      });
+    }
+  });
+  array.push({
+    result: {
+      name: i.name + "=" + j.name,
+      latitude: i.latitude + "=" + j.latitude,
+      longitude: i.longitude + "=" + j.longitude,
+      address1: i.address1 + "=" + j.address1,
+      address2: i.address2 + "=" + j.address2
+    }
+  });
+};
+
+
+// analyze all search results to find locations which meet conditions
+var analyzeResult = function(displayedArray) {
+  var analyzeArray0 = [];
+  var analyzeArray1 = [];
+  var analyzeArray2 = [];
+  var analyzeArray3 = [];
```

Comment:
What if you have more than 4 `analyzeArray` to analyze. My suggestion is don't hardcode this the variable like this.